Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -2379,24 +2381,21 @@ void WebExtension::populateCommandsIfNeeded()
     }
 }
 
-std::optional<WebExtension::DeclarativeNetRequestRulesetData> WebExtension::parseDeclarativeNetRequestRulesetObject(const JSON::Object& rulesetObject, RefPtr<API::Error>& error)
+Expected<std::optional<WebExtension::DeclarativeNetRequestRulesetData>, RefPtr<API::Error>> WebExtension::parseDeclarativeNetRequestRulesetObject(const JSON::Object& rulesetObject)
```

Comment:
Shouldn't need optional anymore. Error can be Ref since it look like it is never null.

```suggestion
Expected<WebExtension::DeclarativeNetRequestRulesetData, Ref<API::Error>> WebExtension::parseDeclarativeNetRequestRulesetObject(const JSON::Object& rulesetObject)
```